src/main.rs

```diff
@@ -240,6 +245,32 @@ fn get_languages_from_env() -> Vec<String> {
     )
 }
 
+fn spawn_editor(custom_pages_dir: &Path, file_name: &str) -> anyhow::Result<()> {
+    let res = create_dir_all(custom_pages_dir)
+        .context(format!("Fail to create dir {:?}", custom_pages_dir));
+    if !custom_pages_dir.is_dir() {
+        res?;
+        return Err(anyhow!("Fail to create dir {:?}", custom_pages_dir));
+    }
```

Comment:
See https://github.com/tealdeer-rs/tealdeer/pull/388#discussion_r1921624380

```suggestion
    create_dir_all(custom_pages_dir).context("Failed to create custom pages directory")?;
```